Libraries/hipSOLVER/sygvj/main.cpp

```diff
@@ -0,0 +1,202 @@
+    HIPSOLVER_CHECK(hipsolverDnXsyevjSetTolerance(syevj_params, tol));
+    // Default value of max. sweeps is 100.
+    HIPSOLVER_CHECK(hipsolverDnXsyevjSetMaxSweeps(syevj_params, max_sweeps));
+
+    // Query and allocate working space.
+    HIPSOLVER_CHECK(hipsolverDsygvj_bufferSize(hipsolver_handle,
+                                               itype,
+                                               jobz,
+                                               uplo,
+                                               n,
+                                               d_A,
+                                               lda,
+                                               d_B,
+                                               ldb,
+                                               d_W,
+                                               &lwork,
+                                               syevj_params));
+    HIP_CHECK(hipMalloc(&d_work, lwork));
+
+    // Compute spectrum (written to d_W) and eigenvectors (writtten to d_A).
+    HIPSOLVER_CHECK(hipsolverDsygvj(hipsolver_handle,
+                                    itype,
+                                    jobz,
+                                    uplo,
+                                    n,
+                                    d_A,
+                                    lda,
+                                    d_B,
+                                    ldb,
+                                    d_W,
+                                    d_work,
+                                    lwork,
+                                    d_sygvj_info,
+                                    syevj_params));
+
+    // Check returned info value.
+    int syevj_info{};
+    HIP_CHECK(hipMemcpy(&syevj_info, d_sygvj_info, sizeof(syevj_info), hipMemcpyDeviceToHost));
+    int errors{};
+    if(syevj_info == 0)
+    {
+        std::cout << "Eigenvalues successfully computed: ";
+
+        // Copy the resulting vector of eigenvalues to the host.
+        std::vector<double> W(n, 0);
+        HIP_CHECK(hipMemcpy(W.data(), d_W, sizeof(double) * n, hipMemcpyDeviceToHost));
+
+        // Print eigenvalues and compare them with the expected values.
+        const std::vector<double> expected_eigenvalues{0.158660256604, 0.370751508101882, 0.6};
+        for(int i = 0; i < n; ++i)
+        {
+            std::cout << W[i] << (i < n - 1 ? ", " : "\n");
+            errors += std::abs(expected_eigenvalues[i] - W[i]) > tol;
+        }
+
+        // Copy the resulting vector of eigenvectors to the host.
+        std::vector<double> V(size_A, 0);
+        HIP_CHECK(hipMemcpy(V.data(), d_A, sizeof(double) * size_A, hipMemcpyDeviceToHost));
+
+        std::cout << "Eigenvectors:" << std::endl;
+        for(int i = 0; i < n; ++i)
+        {
+            std::cout << "{ ";
+            for(int j = 0; j < n; ++j)
+            {
+                std::cout << V[i * n + j] << (j < n - 1 ? ", " : " }\n");
```

Comment:
Here's another place to use lda instead of n

Reply:
Fixed!